docs/design-documents/20230228-connector-lifecycle-methods.md

```diff
@@ -0,0 +1,238 @@
+# Connector Lifecycle Methods
+
+## Introduction
+
+Connector Lifecycle Methods are additional integration points that would let
+connectors execute certain operations when the connector is first created, when
+its configuration is updated or when it gets deleted.
+
+### Background
+
+We first saw a need for this in the Postgres connector, which can create and
+manage its own logical replication slot. We run into a cleanup problem - if the
+logical replication slot is created by the connector it must also be deleted by
+the connector, so the connector currently deletes it in `Teardown`. This defeats
```

Comment:
to clarify, `Teardown` currently happens when stopping the pipeline? and not only on the deletion of the connector?

Reply:
`Teardown` happens _only_ when stopping the pipeline, on connector deletion nothing happens right now.

Reply:
sounds good, thanks for clarifying!